### PR TITLE
Rename component to SingletonManager

### PR DESCRIPTION
**Pull request by Vatyx**
Renamed the component from `GlobalStateManager` to `SingletonManager` to be more descriptive.

### CODE REVIEW

File: SpatialGDK/Extras/schema/global_state_manager.schema [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 package unreal;
 
-component GlobalStateManager {
+component SingletonManager {
```
Comment (improbable-valentyn) [inline SpatialGDK/Extras/schema/global_state_manager.schema]: Shouldn't we also change the filename as well as the class name?

Comment (joshuahuburn) [inline SpatialGDK/Extras/schema/global_state_manager.schema]: Docs updates in generating-a-snapshot with this PR?

Comment (joshuahuburn) [reply SpatialGDK/Extras/schema/global_state_manager.schema]: And the `GLOBAL_STATE_MANAGER_COMPONENT_ID` in spatial constants

Comment (joshuahuburn) [reply SpatialGDK/Extras/schema/global_state_manager.schema]: Actually why not just `Singletons`

**Pull request merged**